viewpump/src/main/java/io/github/inflationx/viewpump/ViewPumpLayoutInflater.java

```diff
@@ -0,0 +1,436 @@
+            super.setFactory2(new WrapperFactory2(factory2));
+        } else {
+            super.setFactory2(factory2);
+        }
+    }
+
+    private void setPrivateFactoryInternal() {
+        // Already tried to set the factory.
+        if (mSetPrivateFactory) return;
+        // Reflection (Or Old Device) skip.
+        if (!ViewPump.get().isReflection()) return;
+        // Skip if not attached to an activity.
+        if (!(getContext() instanceof Factory2)) {
+            mSetPrivateFactory = true;
+            return;
+        }
+
+        final Method setPrivateFactoryMethod = ReflectionUtils.getMethod(LayoutInflater.class, "setPrivateFactory");
+
+        if (setPrivateFactoryMethod != null) {
+            ReflectionUtils.invokeMethod(this,
+                    setPrivateFactoryMethod,
+                    new PrivateWrapperFactory2((Factory2) getContext(), this));
+        }
+        mSetPrivateFactory = true;
+    }
+
+    // ===
+    // LayoutInflater ViewCreators
+    // Works in order of inflation
+    // ===
+
+    /**
+     * The Activity onCreateView (PrivateFactory) is the third port of call for LayoutInflation.
+     * We opted to manual injection over aggressive reflection, this should be less fragile.
+     */
+    @Override
+    @TargetApi(Build.VERSION_CODES.HONEYCOMB)
+    public View onActivityCreateView(View parent, View view, String name, Context context, AttributeSet attrs) {
+        return ViewPump.get().inflate(InflateRequest.builder()
+                .name(name)
+                .context(context)
+                .attrs(attrs)
+                .parent(parent)
+                .fallbackViewCreator(new ActivityViewCreator(this, view))
+                .build()).view();
+    }
+
+    /**
+     * The LayoutInflater onCreateView is the fourth port of call for LayoutInflation.
+     * BUT only for none CustomViews.
+     */
+    @Override
+    @TargetApi(Build.VERSION_CODES.HONEYCOMB)
+    protected View onCreateView(View parent, String name, AttributeSet attrs) throws ClassNotFoundException {
+        return ViewPump.get().inflate(InflateRequest.builder()
+                .name(name)
+                .context(getContext())
+                .attrs(attrs)
+                .parent(parent)
+                .fallbackViewCreator(parentAndNameAndAttrsViewCreator)
+                .build()).view();
+    }
+
+    /**
+     * The LayoutInflater onCreateView is the fourth port of call for LayoutInflation.
+     * BUT only for none CustomViews.
+     * Basically if this method doesn't inflate the View nothing probably will.
+     */
+    @Override
+    protected View onCreateView(String name, AttributeSet attrs) throws ClassNotFoundException {
+        return ViewPump.get().inflate(InflateRequest.builder()
+                .name(name)
+                .context(getContext()) // TODO: is this OK? before was fallbackView.getContext()
```

Comment:
Actually, I looked into this further, and it appears as though previously, when `LayoutInflater.createView(name, prefix, attrs)` was called, it used the `mContext` from the `LayoutInflater`, so by using `getContext()`, which returns `mContext`, I am using the same `Context` [[link to source](https://android.googlesource.com/platform/frameworks/base/+/refs/heads/master/core/java/android/view/LayoutInflater.java#609)]